### ISSUE TITLE
Reporting "The following file(s) do not belong to any projects" for Non-Existent E2E Files

### ISSUE THREAD

**Issue by PapaStef** [nrwl/nx#14762]
### Current Behavior

NX is failing to track deleted projects. When generating a React application + associated E2E project, then deleting the files and all refs, it seems to think the original files are still there and throws a linter exception.

### Expected Behavior

NX somehow seems to be caching or storing some internal reference to files that no longer exist. It should not have mismatched state like that.

### Steps to Reproduce

1. Run the React:app generator for a project named `my-test`
2. Completely delete files from workspace and eliminate all code references
3. Run workspace linting with `npx nx workspace-lint`

```
 >  NX   The following file(s) do not belong to any projects:

   - apps/my-test-e2e/cypress.config.ts
   - apps/my-test-e2e/project.json
   - apps/my-test-e2e/src/e2e/app.cy.ts
   - apps/my-test-e2e/src/fixtures/example.json
   - apps/my-test-e2e/src/support/app.po.ts
```

### Nx Report

```shell
Node : 16.18.1
   OS   : darwin x64
   npm  : 8.19.2

   nx : 15.0.12
   @nrwl/angular : Not Found
   @nrwl/cypress : 15.0.12
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.0.12
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.0.12
   @nrwl/expo : Not Found
   @nrwl/express : 15.0.12
   @nrwl/jest : 15.0.12
   @nrwl/js : 15.0.12
   @nrwl/linter : 15.0.12
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : 15.0.12
   @nrwl/nx-cloud : 15.0.2
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 15.0.12
   @nrwl/react-native : Not Found
   @nrwl/rollup : 15.0.12
   @nrwl/schematics : Not Found
   @nrwl/storybook : 15.0.12
   @nrwl/web : 15.0.12
   @nrwl/webpack : 15.0.12
   @nrwl/workspace : 15.0.12
   typescript : 4.9.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
```

**Comment by PapaStef** [comment]
For anyone else who stumbles on this, I fixed it by clearing the NX cache and restarting the daemon

```sh
> npx nx reset
> npx nx daemon --start
```

**Comment by AgentEnder** [comment]
This will be fixed in 15.7.0, since workspace-lint is being removed.
